activesupport/test/rescuable_test.rb

```diff
@@ -57,6 +57,18 @@ def ronanize
     raise MadRonon.new("dex")
   end
 
+  def more_specific_cause
+    ronanize
+  rescue
+    nuke
+  end
+
+  def less_specific_cause
+    nuke
+  rescue
+    ronanize
+  end
+
```

Comment:
For testing which rescue handler is chosen: the exception's, or the exception cause's